src/number_theory/function_field.lean

```diff
@@ -107,4 +109,119 @@ is_integral_closure.is_dedekind_domain Fq[X] (ratfunc Fq) F _
 
 end ring_of_integers
 
+/-! ### The place at infinity on Fq(t) -/
+
+section infty_valuation
+
+variable [decidable_eq (ratfunc Fq)]
+
+/-- The valuation at infinity is the nonarchimedean valuation on `Fq(t)` with uniformizer `1/t`.
+Explicitly, if `f/g ∈ Fq(t)` is a nonzero quotient of polynomials, its valuation at infinity is
+`multiplicative.of_add(degree(f) - degree(g))`. -/
+def infty_valuation_def (r : ratfunc Fq) : with_zero (multiplicative ℤ) :=
+ite (r = 0) 0 (multiplicative.of_add ((r.num.nat_degree : ℤ) - r.denom.nat_degree))
```

Comment:
```suggestion
if r = 0 then 0 else multiplicative.of_add ((r.num.nat_degree : ℤ) - r.denom.nat_degree))
```

Reply:
@eric-wieser, is this the preferred syntax? Are these two definitions equivalent, or is there any difference between using `ite` and `if-then-else` that I should be aware of? Thanks!